code/game/objects/items/devices/radio/headset.dm

```diff
@@ -66,6 +67,8 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 /obj/item/radio/headset/talk_into(mob/living/M, message, channel, list/spans, datum/language/language, list/message_mods)
 	if (!listening)
 		return ITALICS | REDUCE_RANGE
+	if(channel == RADIO_CHANNEL_UPLINK)
+		SEND_SIGNAL(src, COMSIG_RADIO_UPLINK_MESSAGE, M, message)
 	return ..()
```

Comment:
I think it would probably be better to intercept a global radio signal, and then check if the channel is the same inside the component.
Also might be an idea to early return before sending the message, not sure if it is possible to listen on the uplink channel for free valid detection.

Reply:
well uh it doesnt actually send a radio message anywhere you just whisper it

Reply:
Could it be possible to build a server on that frequency so it does go?

Reply:
i have no idea how to do that